modules/vm/templates/entrypoint.sh.tpl

```diff
@@ -14,9 +14,102 @@ done
 # Login in Azure CLI with managed identity (user or system assigned)
 az login --identity
 
+# Get the license
+az keyvault secret download --vault-name ${key_vault_name} --name graphdb-license --file /etc/graphdb/graphdb.license --encoding base64
+
 # TODO: Find/create/mount volumes
 # https://learn.microsoft.com/en-us/azure/virtual-machine-scale-sets/tutorial-use-disks-cli
-
+# Find/create/attach volumes
+
+instanceHostname=\'$(hostname)\'
+subscriptionID=$(az account show --query "id" --output tsv)
+resourceGroup=$(az vmss list --query "[0].resourceGroup" --output tsv)
+vmssName=$(az vmss list --query "[0].name" --output tsv)
+instanceID=$(az vmss list-instances --resource-group $resourceGroup --name $vmssName --query "[?contains(osProfile.computerName, $${instanceHostname})].instanceId" --output tsv)
+zoneID=$(az vmss list-instances --resource-group $resourceGroup --name $vmssName --query "[?contains(osProfile.computerName, $${instanceHostname})].zones" --output tsv)
+regionID=$(az vmss list-instances --resource-group $resourceGroup --name $vmssName --query "[?contains(osProfile.computerName, $${instanceHostname})].location" --output tsv)
+# TODO replace with external variables
+tier="P40"
+lun=2 # Based on this we find and mount the disk in the VM
+
+diskSizeGB=128
+
+# TODO Define the disk name based on the hostname ??
+diskName="Disk_$${vmssName}_$${instanceID}"
```

Comment:
In case of VM failure, and when it gets recreated - wouldn't the ID be different? This might cause slight confusion.